python/ray/tune/examples/fault_tolerant_function_example.py

```diff
@@ -0,0 +1,92 @@
+import argparse
+import cloudpickle
+import numpy as np
+import time
+
+import ray
+from ray import tune
+from ray.tune import track
+
+
+class OptimusFn(object):
+    """Simulate learning."""
+
+    def __init__(self, params, max_t=10000):
+        self.params = params
+        self.noise = np.random.normal(size=max_t) * 0.005
+
+    def __call__(self, k, add_noise=True):
+        time.sleep(0.5)
+        b0, b1, b2 = self.params
+        score = (b0 * k / 100 + 0.1 * b1 + 0.5)**(-1) + b2 * 0.01
+        if add_noise:
+            return score + abs(self.noise[k])
+        else:
+            return score
+
+
+CHECKPOINT_FREQ = 10
+TRAINING_ITERS = 1000
+
+
+def train(config):
+    if track.is_pending_restore():
```

Comment:
Why would anyone ever want to do this instead of using the class-based trainable?

Reply:
in some scenarios this is more convenient. i think many users don't want to use the class-based api since it requires non-trivial changes to existing code

Reply:
I see, I guess that makes sense. Just to clarify, I have nothing against supporting this, but was just curious why people would want that.